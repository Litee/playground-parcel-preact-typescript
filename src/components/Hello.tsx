import { h, Component } from 'preact';

export default class Hello extends Component<HelloProps, HelloState> {
	clickHandler: () => void;

	constructor() {
		super();
		this.state.counter = 0;
		this.clickHandler = () => {
			this.setState({counter : this.state.counter + 1});
		}
	}

	render(props: HelloProps, state: HelloState) {
		return <a class="greeting" onClick={this.clickHandler} href="#">Hello, {props.name}! You clicked me {state.counter} times.</a>;
	}
}

export interface HelloProps {
	name: string;
}

export interface HelloState {
	counter: number;
}