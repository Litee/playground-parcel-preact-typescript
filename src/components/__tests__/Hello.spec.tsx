import { h } from 'preact';
import { shallow } from 'preact-render-spy';
import Hello from '../Hello';

describe('Hello component', () => {
	it('should render correctly', () => {
		const context = shallow(
			<Hello name="John Doe" />
		);
		expect(context).toMatchSnapshot();
	});

	it('should process click', () => {
		const context = shallow(
			<Hello name="John Doe" />
		);
		context.find('[onClick]').simulate('click');
		expect(context).toMatchSnapshot();
	});
});