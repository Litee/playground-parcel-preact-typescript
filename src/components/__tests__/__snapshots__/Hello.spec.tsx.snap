// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello component should process click 1`] = `
preact-render-spy (1 nodes)
-------
<a
  class="greeting"
  onClick={[Function anonymous]}
  href="#"
>
  Hello, John Doe! You clicked me 1 times.
</a>

`;

exports[`Hello component should render correctly 1`] = `
preact-render-spy (1 nodes)
-------
<a
  class="greeting"
  onClick={[Function anonymous]}
  href="#"
>
  Hello, John Doe! You clicked me 0 times.
</a>

`;
